test/e2e/app-dir/metadata/metadata.test.ts

```diff
@@ -855,4 +855,21 @@ describe('app dir - metadata', () => {
       )
     }
   })
+
+  it('regression: renders a large shell', async () => {
+    const pageErrors: unknown[] = []
+    await next.browser('/large-shell/foo', {
+      beforePageLoad(page) {
+        page.on('pageerror', (error) => {
+          pageErrors.push(error)
+        })
+      },
+    })
+
+    // TODO: Assert on errorless pages by default.
+    // This isn't 100% accurate.
+    // We sometimes receive the pageerror after the hydration complete event
+    // since that event is just for shell hydration not everything being hydrated.
+    expect(pageErrors).toEqual([])
```

Comment:
To make it 100% accurate, what if we add a client component to the page that logs in an effect, and wait for the browser log to appear before continuing with this assertion? By the way, do we even need to collect the page errors? I think it will just throw if there is a page error.

Reply:
It doesn't throw if there is a pageerror. pageerror is just `window.onError`.

Reply:
>To make it 100% accurate, what if we add a client component to the page that logs in an effect, and wait for the browser log to appear before continuing with this assertion?

I tried that but it didn't help. There might be a timing issue where we receive pagerror events delayed due to some internal buffering of events by Playwright?